File: leetcode/2363.合并相似的物品.cpp
class Solution {
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2) {
        map<int,int> mp;
        vector<vector<int>> res;
        for(auto a:items1){
            mp[a[0]]+=a[1];
        }
        for(auto a:items2){
            mp[a[0]]+=a[1];
        }
        for(auto [x,y] :mp){
            res.push_back({x,y});
        }
        return res;
    }
};

File: leetcode/2363.合并相似的物品.py
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
        s = Counter()
        for x,y in items1:
            s[x] += y
        for x,y in items2:
            s[x] += y
        return sorted(s.items())
        